src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,343 @@
+import * as React from 'react';
+import { If, Then, Else } from 'react-if';
+import * as _ from 'lodash';
+import {observer} from "mobx-react";
+import {
+    IClinicalGroupMatch, IGenomicGroupMatch, IGenomicMatch, ITrial,
+    ITrialMatch, IDetailedTrialMatch, IArmMatch
+} from "../../../shared/model/MatchMiner";
+import styles from './style/trialMatch.module.scss';
+import { computed } from "mobx";
+import LazyMobXTable from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import SampleManager from "../sampleManager";
+import DefaultTooltip, { placeArrowBottomLeft } from "../../../shared/components/defaultTooltip/DefaultTooltip";
+
+export type ITrialMatchProps = {
+    sampleManager?: SampleManager | null;
+    trials: Array<ITrial>;
+    trialMatches:Array<ITrialMatch>;
+    containerWidth: number;
+}
+
+export type ITrialMatchState = {
+    detailedTrialMatches: Array<IDetailedTrialMatch>;
+}
+
+enum ColumnKey {
+    ID = 'ID',
+    TITLE = 'Title',
+    MATCHINGCRITERIA = 'Matching Criteria'
+}
+
+class TrialMatchTableComponent extends LazyMobXTable<IDetailedTrialMatch> {
+
+}
+
+@observer
+export default class TrialMatchTable extends React.Component<ITrialMatchProps, ITrialMatchState> {
+
+    constructor(props: ITrialMatchProps) {
+        super(props);
+        this.state = { detailedTrialMatches: this.buildDetailedTrialMatches()};
+    }
+
+    @computed
+    get columnsWidth() {
+        return {
+            [ColumnKey.ID]: 140,
+            [ColumnKey.TITLE]: 0.35 * (this.props.containerWidth - 140),
+            [ColumnKey.MATCHINGCRITERIA]: 0.65 * (this.props.containerWidth - 140)
+        };
+    }
+
+    private _columns = [{
+        name: ColumnKey.ID,
+        render: (trial: IDetailedTrialMatch) => (
+            <div>
+                <div><a target="_blank" href={"https://www.mskcc.org/cancer-care/clinical-trials/" + trial.protocolNo}>{trial.protocolNo}</a></div>
```

Comment:
For cases that don't have protocalNo or NctId, we need to handle the case.

Reply:
Added <If></If> to create HTML element for valid nctId and protocolNo.